### PR TITLE
ddl: fix Error in accuracy when adding CURRENT_TIMESTAMP column

### PR DESCRIPTION
**Pull request by coolboydan**
###  What problem does this PR solve?
Issue Number: #23784

Problem Summary:

### 1. Minimal reproduce step (Required)
create table test(id int,update_time datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6));
insert into test(id) values (1);
select * from test;
alter table test add column update_time1 datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6) ;
select * from test;

### 2. What did you expect to see? (Required)
mysql> select * from test;
+------+----------------------------+----------------------------+
| id | update_time | update_time1 |
+------+----------------------------+----------------------------+
| 1 | 2021-03-31 19:13:15.441962 | 2021-03-31 19:13:28.381179 |
+------+----------------------------+----------------------------+
1 row in set (0.01 sec)

### 3. What did you see instead (Required)
mysql> select * from test;
+------+----------------------------+----------------------------+
| id | update_time | update_time1 |
+------+----------------------------+----------------------------+
| 1 | 2021-03-31 19:13:15.441962 | 2021-03-31 19:13:28.000000 |
+------+----------------------------+----------------------------+
1 row in set (0.01 sec)

### What is changed and how it works?
What's Changed:
repair Error in accuracy when adding CURRENT_TIMESTAMP column.
How it Works:
Allows formatting to specified precision when CURRENT_TIMESTAMP adds columns. 

### Related changes

- Need to cherry-pick to the release branch
   cherry-pick: V3.0,V3.1,V4.0

### Check List
Tests

- Unit test

### Release note
- fix Error in accuracy when adding CURRENT_TIMESTAMP column.

### CODE REVIEW

File: ddl/column.go [hunk 1]
```diff
@@ -1726,11 +1726,20 @@ func generateOriginDefaultValue(col *model.ColumnInfo) (interface{}, error) {
...
 	}
 
+	timeFormat := types.TimeFormat
+	if col.Decimal != 0 {
+		pend := "."
+		for i := 0; i < col.Decimal; i++ {
+			pend += "9"
+		}
+		timeFormat = types.TimeFormat + pend
+	}
```
Comment (zimulala) [inline ddl/column.go:1736]: Could you use `types.TimeFSPFormat` directly?

**Comment by bb7133** [comment]
Thanks for your contribution!

**Comment by zimulala** [comment]
Please add some tests for this PR.

**Comment by bb7133** [comment]
Please add some test cases.

File: ddl/column.go (hunk 1 above)
Comment (coolboydan) [reply ddl/column.go:1736]: Cannot use TimeFSPFormat, this is the specified precision. The user is going to use CURRENT_TIMESTAMP of different precision, like 3,4,5. Inconsistency with MySQL results needs to be fixed.

Comment (coolboydan) [reply ddl/column.go]: 好主意

Comment (coolboydan) [reply ddl/column.go]: Good idea。

**Comment by xiongjiwei** [comment]
any update?

**Comment by wjhuang2016** [comment]
@coolboydan ping, could you update this PR?

**Comment by Defined2014** [comment]
Already fixed by #27149

**Pull request closed without merging**